src/components/attach/AttachHub.js

```diff
@@ -0,0 +1,406 @@
+        const fileObj = newFiles[i];
+        const formData = new FormData();
+        formData.append("file", fileObj, fileObj.name.toLocaleLowerCase());
+        let file = formData.get("file");
+
+        if (file.name.length > 255) {
+          throw new Error("The file name should not exceed 255 characters.", {
+            message: "The file name should not exceed 255 characters.",
+          });
+        }
+        if (file.size > 104857600) {
+          throw new Error("The file size should not exceed 100 MB.", {
+            message: "The file size should not exceed 100 MB.",
+          });
+        }
+
+        const fileExtension = file.name.split(".").slice(-1)[0];
+
+        if (
+          !globalConstants.allowedFileFormats.includes(file.type) &&
+          !["heic", "HEIC"].includes(fileExtension)
+        ) {
+          throw new Error("Please select an image file.", {
+            message: "Please select an image file.",
+          });
+        } else if (["heic", "HEIC"].includes(fileExtension)) {
+          const tmp = await heicToPng(file);
+          const pngFile = await compressAndHashFile(tmp);
+
+          selectedFiles.push(pngFile);
+          continue;
+        }
+
+        if (file.type.startsWith("image/")) {
+          file = await compressAndHashFile(file);
+        }
+
+        if (file.type.startsWith("video/")) {
+          file.localUrl = URL.createObjectURL(file);
+        }
+        selectedFiles.push(file);
+      }
+
+      if (files?.length + newFiles.length > 10) {
+        throw new Error("The maximum limit for file uploads is 10.", {
+          message: "The maximum limit for file uploads is 10.",
+        });
+      }
+    } catch (err) {
+      err.message
+        ? showCustomAlert(err.message, "warning")
+        : console.error(err);
+      return;
+    }
+
+    setFiles((prevFiles) => [...prevFiles, ...selectedFiles]);
+    setIsPending(false);
+  };
+
+  const uploadInputText = useCallback(() => {
+    if (inputTextRef.current.value) {
+      localStorage.setItem("mtext", inputTextRef.current.value);
+      inputTextRef.current.value = "";
+    }
+  }, [inputTextRef]);
+
+  const syncInputText = useCallback(() => {
+    const mtext = localStorage.getItem("mtext");
+    if (mtext) {
+      localStorage.removeItem("mtext");
+      inputTextRef.current.value = mtext;
+    }
+  }, [inputTextRef]);
+
+  const closeModal = useCallback(
+    (isForseClose) => {
+      const close = () => {
+        removeAndNavigateLastSection(pathname + hash);
+        uploadInputText();
+        return true;
+      };
+
+      if (isForseClose === true) {
+        return close();
+      }
+
+      if (
+        files.length &&
+        !window.confirm("Are you sure you want to cancel sending files?")
+      ) {
+        return false;
+      }
+      return close();
+    },
+    [files.length, pathname, hash, uploadInputText]
+  );
+
+  const removeFile = useCallback(
```

Comment:
any strong reason for wrapping everything into useCallback ?

I feel we can define removeFile/closeModal/syncInputText/uploadInputText as regular functions